checks/restic.go

```diff
@@ -0,0 +1,104 @@
+package checks
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/flanksource/canary-checker/api/external"
+	v1 "github.com/flanksource/canary-checker/api/v1"
+	"github.com/flanksource/canary-checker/pkg"
+	"github.com/flanksource/commons/exec"
+	"github.com/flanksource/commons/logger"
+	osExec "os/exec"
+	"time"
+)
+
+func init() {
+	//register metrics here
+}
+
+const (
+	resticPasswordEnvKey       = "RESTIC_PASSWORD"
+	resticAwsAccessKeyIdEnvKey = "AWS_ACCESS_KEY_ID"
+	resticAwsSecretAccessKey   = "AWS_SECRET_ACCESS_KEY"
+)
+
+type ResticChecker struct{}
+
+func (c *ResticChecker) Type() string {
+	return "restic"
+}
+
+func (c *ResticChecker) Run(config v1.CanarySpec) []*pkg.CheckResult {
+	var results []*pkg.CheckResult
+	for _, conf := range config.Restic {
+		results = append(results, c.Check(conf))
+	}
+	return results
+}
+
+func (c *ResticChecker) Check(extConfig external.Check) *pkg.CheckResult {
+	logger.Debugf("Starting Restic checks")
+	start := time.Now()
+	resticCheck := extConfig.(v1.ResticCheck)
+	envVars := getEnvVars(resticCheck)
+	if err := checkIntegrity(resticCheck.Repository, resticCheck.CaCert, envVars); err != nil {
+		return Failf(resticCheck, "failed to check integrity of repository %v", err)
+	}
+	logger.Debugf("Restic integrity check succeeded for repository: %s", resticCheck.Repository)
+	if err := checkBackupFreshness(resticCheck.Repository, resticCheck.MaxAge, resticCheck.CaCert, envVars); err != nil {
+		return Failf(resticCheck, "failed while checking backup freshness: %v", err)
```

Comment:
This shouldn't return an error, but rather a message with how much older it is than than the maxage

Reply:
Okay... I can include how much the backup is older than the maxAge but the canary should fail in that case, right?
So Only the  return message needs to be adjusted right?
The `Failf` remains here, if I got it correctly?

Reply:
Correct